src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -455,67 +575,208 @@ impl<'a> ImmutableBuffer<'a> {
         todo!()
     }
 
-    fn read_e_expression(self, opcode: Opcode) -> IonResult<RawBinaryEExpression_1_1<'a>> {
+    #[inline]
+    pub fn read_e_expression(self, opcode: Opcode) -> ParseResult<'a, BinaryEExpression_1_1<'a>> {
         use OpcodeType::*;
-        let (macro_id, buffer_after_id) = match opcode.opcode_type {
-            EExpressionWithAddress => (
-                MacroIdRef::LocalAddress(opcode.byte as usize),
-                self.consume(1),
-            ),
+        match opcode.opcode_type {
+            EExpressionWithAddress => return self.read_eexp_with_address_in_opcode(opcode),
             EExpressionAddressFollows => todo!("e-expr with trailing address; {opcode:#0x?}",),
+            EExpressionWithLengthPrefix => return self.read_eexp_with_length_prefix(opcode),
             _ => unreachable!("read_e_expression called with invalid opcode"),
         };
+    }
 
-        // TODO: When we support untagged parameter encodings, we need to use the signature's
-        //       parameter encodings to drive this process. For now--while everything is tagged
-        //       and cardinality is always required--we just loop `num_parameters` times.
-        let macro_def = self
```

Comment:
The `ImmutableBuffer` and `TextBufferView` types both need a reference to the bump allocator in order to:
* (text) Decode escape sequences
* (text) Decode base64
* (text) Join multi-section long-form strings into one value.
* (any) Bump-allocate large data types so we can pass around a `&'top` reference instead.
* (any) Bump-allocate macro-constructed values from `(make_string)`, `(make_list)`, `(annotate)`, etc.

Starting with this PR, the buffers need to be able to parse macro invocations too. That means that all parsing methods would need to accept a reference to the macro table in order to successfully parse a value which could have a macro invocation at any level of nesting. (Including: `read`, `read_sequence_value`, `peek_sequence_value`, `read_struct`, `read_struct_field`, etc etc.) Importantly, data types that would lazily call a parsing method (`LazyValue`, `LazyList`, `LazyStruct`, .... `LazyExpandedValue`, `LazyExpandedList` .... `LazyRawValue`, `LazyRawList`, etc) would also need to hold/track a reference to the macro table for the moment they eventually decided to call a parsing method.

Rather than selectively plumb through distinct `allocator` and `macro_table` references throughout, I chose to simply give each buffer a reference to the encoding context. By definition, the encoding context is 'frozen' throughout the codebase when parsing occurs, so there's not much downside to liberally handing out references to it.